spec/features/user_changes_list_records_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+
+feature "user changes list records" do
+  scenario "updates successfully" do
+    user = create(:user)
+    list = create(:list, user: user)
+
+    visit list_path(list, as: user)
+    click_on "Edit"
+
+    fill_form(
+      "List",
+      name: "New List Name",
+    )
+    click_on "Save changes"
+
+    expect(page).to have_list("New List Name")
+  end
+
+  scenario "cancel edits" do
+    user = create(:user)
+    list = create(:list, name: "List", user: user)
+
+    visit list_path(list, as: user)
+    click_on "Edit"
+
+    fill_form(
+      "List",
+      name: "New List Name",
+    )
+    click_on "Cancel"
+
+    expect(page).to have_list("List")
+  end
+
+  scenario "deletes list" do
+    user = create(:user)
+    create(:list, name: "First list", user: user)
+    list = create(:list, name: "List to delete", user: user)
+
+    visit list_path(list, as: user)
+    click_on "Delete"
+
+    expect(page).to have_list("First list")
+    expect(page).to_not have_list("List to delete")
+  end
+
+  def have_list(name)
```

Comment:
I think this has been used a couple different times across various tests, how about moving it out to a shared file?